### PR TITLE
convert model to keras format

### PR DESCRIPTION
**Pull request by junzis**
Here is a summary of the updates:

- The `saved_model` model files are outdated for newer tensorflow versions. I converted it to the recommended `keras` format and updated the model loading code.
- Python 3.8 is also almost end-of-life. So I recommend a newer version in the readme.
- Some minor code formatting for readability (auto-formated with `black`).
- I added a `.gitignore` file to not track common cached files from python.

### CODE REVIEW

**Comment by JarryGabriel** [comment]
Hi @junzis thank you so much, this is great ! Just a question regarding the keras format, is it still optimized for inference ? What I remember is that the save_model basically remove all the gradient computations and therefore was way faster during inference.

**Comment by junzis** [comment]
This seems to be true. The inference speed from saved_model is about 4x faster than the Keras model. 
I just did a test, and it's about 40ms vs 150ms for the text trajectory.

**Comment by JarryGabriel** [comment]
I just had look through Tensorflow 2.16.1 doc, it seems that is still exists but maybe then changed the repository format ?

 https://www.tensorflow.org/api_docs/python/tf/saved_model/load

Other possibility, it was created with a python 3.8 and then is not working properly with 3.11. What error did you receive when running the code ?

**Comment by junzis** [comment]
I noticed that I was actually using a lower version of TensorFlow with my uv pip environment...😅
I will close this PR, and make a new one later.
